docs/src/pages/nuxt.mdx

```diff
@@ -0,0 +1,257 @@
+      console.log("uploaded with the following metadata:", metadata);
+      console.log("files successfully uploaded:", file);
+    }),
+
+  withoutMdwr: f({
+    image: {
+      maxFileCount: 2,
+      maxFileSize: "16MB",
+    },
+  })
+    .middleware(() => {
+      return { testMetadata: "lol" };
+    })
+    .onUploadComplete(({ metadata, file }) => {
+      console.log("uploaded with the following metadata:", metadata);
+      console.log("files successfully uploaded:", file);
+    }),
+} satisfies FileRouter;
+
+export type OurFileRouter = typeof uploadRouter;
+```
+
+### Create an API route using the FileRouter
+
+<Callout type="warning">
+  Note: This is the ONLY FILE WHERE THE PATH MATTERS. You need to serve this API
+  from `/api/uploadthing` on your application.
+</Callout>
+
+```ts copy filename="server/api/uploadthing.ts"
+import { createNuxtRouteHandler } from "uploadthing/nuxt";
+
+import { uploadRouter } from "../uploadthing";
+
+export default createNuxtRouteHandler({
+  router: uploadRouter,
+});
+```
+
+### Import CSS
+
+You need to import our styles for the button and dropzone to look right
+
+```ts copy filename="nuxt.config.ts"
+export default defineNuxtConfig({
+  css: ["@uploadthing/vue/style.css"],
+});
+```
+
+### Use the FileRouter in your app
+
+The `@uploadthing/nuxt` package includes "useUploader" builder that you can use
+to generate type-safe component, drop it into app and start uploading files.
+
+```vue copy filename="components/Home.vue"
+<script setup lang="ts">
+import type { OurFileRouter } from "server/uploadthing";
+
+import { useUploader } from "@uploadthing/vue";
+
+const Uploader = useUploader<OurFileRouter>();
+</script>
+
+<template>
+  <main
+    class="flex min-h-screen flex-col items-center justify-center gap-16 p-24"
+  >
+    <Uploader :config="{ endpoint: 'videoAndImage', onClientUploadComplete={()
+    => { alert("Upload Completed"); }} onUploadError={(error: Error) => { // Do
+    something with the error. alert(`ERROR! ${error.message}`); }} }" />
+  </main>
+</template>
+```
+
+</Steps>
+
+### Use the FileRouter in your app
+
+#### Uploader
+
+The `@uploadthing/nuxt` package includes "useUploader" builder that you can use
+to generate type-safe component, drop it into app and start uploading files.
+
+```vue copy filename="components/Home.vue"
+<script setup lang="ts">
+import type { OurFileRouter } from "server/uploadthing";
+
+import { useUploader } from "@uploadthing/vue";
+
+const Uploader = useUploader<OurFileRouter>();
+</script>
+
+<template>
+  <main
+    class="flex min-h-screen flex-col items-center justify-center gap-16 p-24"
+  >
+    <ClientOnly fallback-tag="div" fallback="Loading...">
+      <Uploader :config="{ endpoint: 'videoAndImage', onClientUploadComplete={()
+      => { alert("Upload Completed"); }} onUploadError={(error: Error) => { //
+      Do something with the error. alert(`ERROR! ${error.message}`); }} }" />
+    </ClientOnly>
```

Comment:
why does this have to be in a client-only block? is it not possible to SSR this and have the file config on first paint?

Reply:
This one have to be fixed. ClientOnly is no longer needed